docs/RFCs/0002-crashtracker-user-visable-log.md

```diff
@@ -0,0 +1,169 @@
+As a structured format, it avoids the ambiguity of standard semi-structured stacktrace formats (as used by e.g. Java, .Net, etc).
+Due to the use of native extensions, it is possible for a single stack-trace to include frames from multiple languages (e.g. python may call C code, which calls Rust code, etc).
+Having a single structured format allows us to work across languages.
+
+## Proposed format
+A natural language description of the proposed json format is given here.
+An example is given in Appendix A, and the schema is given in Appendix B.
+
+### Required fields
+- `error`:
+    A structure of type `ErrorData`, described below.
+- `incomplete`:
+    Boolean `false` if the crashreport is complete (i.e. contains all intended data), `true` if there is expected missing data.
+    This can happen becasue the crashtracker is architected to stream data to an out of process receiver, allowing a partial crash report to be emitted even in the case where the crashtracker itself crashed during stack trace collection.
+    This MUST be set to `true` if any required field is missing.
+- `metadata`:
+    Metadata about the system in which the crash occurred:
+    - `library_name`
+    - `library_version`
+    - `family`
+    - `tags`:
+      A set of key:value pairs, representing any tags the crashtracking system wishes to associate with the crash.
+      Examples would include "hostname", "service", and any configuration information the system wishes to track.
+- `os_info`: 
+    The OS + processor architecture on which the crash occurred.
+- `stacktrace`: 
+    This represents the stack of the crashing thread.
+    See below for more details on how stacktraces are formatted.
+- `timestamp`:
+    The time at which the crash occurred, in ISO 8601 format.
+- `uuid`:
+    A UUID v4 which uniquely identifies the crash.
+- `version_id`:
+    An integer version id.
+
+### ErrorData
+- `threads`:
+    This **optional** field contains an array of `Thread` objects.
+    In a multi-threaded program, the collector SHOULD collect the stacktraces of all active threads, and report them here.
+    A `Thread` object has the following fields:
+    - `crashed`: a boolean which tells if the thread crashed.
+    - `name`: Name of the thread (e.g. 'Thread 0').
+    - `stack`: The `StackTrace` of the thread
+    - `state`: **Optional**. Platform-specific state of the thread when its state was captured (CPU registers dump for iOS, thread state enum for Android, etc.).
+    See below for more details on how stacktraces are formatted.
+- `is_crash`:
+    Boolean true if the error was a crash, false otherwise.
+- `kind`:
+    The kind of error that occurred.
+    For a crash, a non-exhaustive set of options include "SigAbort", "SigBus", "SigSegv".
+    This field MAY be extended to include options such at "Panic", "UnhandledException" etc.
+- `message`:
+    A human readable string containing an error message associated with the stack trace.
+- `source_type`:
```

Comment:
It would be nice to have this information at the top level in telemetry payload. I noticed that there is a [Origin](https://github.com/DataDog/dd-go/blob/prod/trace/apps/tracer-telemetry-intake/telemetry-payload/header.go#L87) field that could be used for this.

Reply:
https://github.com/DataDog/libdatadog/issues/727